pydra/engine/specs.py

```diff
@@ -451,9 +451,14 @@ def collect_additional_outputs(self, inputs, output_dir, outputs):
             input_value = getattr(inputs, fld.name, attr.NOTHING)
             if input_value is not attr.NOTHING:
                 if TypeParser.contains_type(FileSet, fld.type):
-                    label = f"output field '{fld.name}' of {self}"
-                    input_value = TypeParser(fld.type, label=label).coerce(input_value)
-                additional_out[fld.name] = input_value
+                    if input_value is not False:
+                        label = f"output field '{fld.name}' of {self}"
+                        input_value = TypeParser(fld.type, label=label).coerce(
+                            input_value
+                        )
+                        additional_out[fld.name] = input_value
+                else:
+                    additional_out[fld.name] = input_value
```

Comment:
This line handles the case where the is a corresponding input with the same name as an output but it isn't of file-type. Is this behaviour intended that the output receives the value of the input field with the same name by default, and therefore should be tested, or just removed?

Reply:
i think it should be removed. I'm not sure what I had in mind...

Reply:
I removed it. We can reinstate it later if a use case for it emerges.